Create user for non-patron class

sakurano92 opened the pull request:
Create user for non-patron class
have been implemented.

sakurano92 committed:
just for me

sakurano92 committed:
Merge remote-tracking branch 'sleepy-hacks/development' into development

sakurano92 committed:
for me

sakurano92 committed:
Merge remote-tracking branch 'sleepy-hacks/development' into development

# Conflicts:
#	app/wama.core.unit.test/Controllers/UserToolControllerTest.cs

sakurano92 committed:
Merge remote-tracking branch 'sleepy-hacks/development' into development

sakurano92 committed:
Merge remote-tracking branch 'sleepy-hacks/development' into development

sakurano92 committed:
Type Patron is now implementable from create user

dubeme commented on app/wama.core/Models/DTOs/UserAccountType.cs:
```diff
@@ -1,13 +1,17 @@
         /// The unknown
         /// </summary>
+        ///
+        [Display(Name = " --- ")]
```
Nice addition

dubeme commented on app/wama.web/Views/Patron/Create.cshtml:
```diff
@@ -1,7 +1,7 @@
 ﻿@{
```
Good start

Although, a couple of question could be asked about improving this form

- [ ] What about having a default selected Account Type, based off the page that requested this from?
- [ ] What about leaving out the Unknown type in the list? _**(since no user account should be that type)**_
- [ ] Currently, when this form is submitted, if there's an error, the error is shown. But all the data that was filled in will be lost. What about retaining those data?

sakurano92 replied:
data doesnt looks like getting lost, it remains in filled even error is shown as far as i testing it

sakurano92 replied:
ill put work on rest of them

dubeme replied:
1st, this comment has moved (I've got no idea why). It belongs to the _**CreateNewUserAccount.cshtml**_ 

These comments, still aren't fully addressed.

sakurano92 replied:
oh yea checked data is getting deleted ill check into it

dubeme replied:
Yikes! Just realized, this file shouldn't be changed at all. Remember, this is the form for creating a patron account from the check in screen. Hence, it should only work for patron accounts. 

Now it makes more sense, why I thought the comment moved. I was supposed to have this for the _**CreateNewUserAccount.cshtml**_. 

This means, that this file should not be changed at all. Please **revert all changes** on this file.

The changes, I suggested in these comments apply to _**CreateNewUserAccount.cshtml**_

dubeme commented on app/wama.web/Views/AdminConsole/UserTool/ViewOneCertification.cshtml:
```diff
@@ -0,0 +1,9 @@
+@model CertificationViewModel
+
+@{ 
+    <div> Work on progress</div>
```
Work in progress

dubeme commented on app/wama.web/Controllers/UserToolController.cs:
```diff
@@ -239,7 +239,13 @@ public IActionResult AddCertification(string memberId)
             return View($"{Constants.ADMIN_CONSOLE_USER_TOOL_DIRECTORY}/ViewCertification.cshtml", Certifications);
         }
-
+        public async Task<IActionResult> ViewOneCertification(string memberId) //need to straight idea first
```
- [ ] What about changing this to, ViewCertification (it still implies you're looking at one certification)?
- [ ] Also, for the parameter, should't be **_certificationID_**?
- [ ] I'm also thinking that the parameter should be taking the **_memberID_** as well (_**since we want to map it down to a certification, that belongs to a user**_) (**_this is optional_**)?

sakurano92 replied:
I was thinking of deleting that function but left it just in case, before i make any change on those, i think i need to talk about that function.
and ill change parameter

sakurano92 replied:
btw changing ViewOneCertification to ViewCertification could cause unwanted bug
since viewing multiple certification is named ViewCertification.
i dont think its good idea to change name of 2 class which one is already used and linked with views.

dubeme replied:
Yeah, that's why we name things consistently. You follow the **_Singular and Plural_** rule.

If a page is used to view all certifications, then it should be something like **_ViewCertifications_** or _**Certifications**_ (note the 's' at the end).
If a page is used to view a single certification, then it should be something like **_ViewCertification_** or _**Certification.**_

dubeme reviewed (changes requested):
Looking good, but not quite. See comments.

sakurano92 committed:
Fix but i dont like it

for selected user from list, line 53, i dont like this method, but serve
the purpose
if any other grammer that i can use to select default value in enum with
<select> tag, let me know

dubeme commented on app/wama.core/Extensions/UserAccountExtensions.cs:
```diff
@@ -47,8 +47,8 @@ public static UserAccountViewModel ToViewModel(this UserAccount user)
 
             switch (user.AccountType)
             {
-                case UserAccountType.Unknown:
```
Nope, do not touch this

sakurano92 replied:
wait you told me to remove unknown since patron account type should not be unknown

dubeme replied:
Yes, but that's not what I meant. Re-read the comments again.

dubeme commented on app/wama.core/Models/DTOs/UserAccountType.cs:
```diff
@@ -1,17 +1,16 @@
         /// <summary>
         /// The unknown
         /// </summary>
-        Unknown,
```
Same here, do not touch this.

dubeme commented on app/wama.web/Controllers/UserToolController.cs:
```diff
@@ -116,6 +116,7 @@ public UserToolController(IUserAccountService userAccountService, ICheckInServic
 
         public IActionResult CreateNewUserAccount(UserAccountType type)
         {
+            ViewBag.AccountType = type;
```
Good, but where is this being used?

dubeme commented on app/wama.web/Views/AdminConsole/UserTool/CreateNewUserAccount.cshtml:
```diff
@@ -44,6 +44,15 @@
+                        asp-items="@Html.GetEnumSelectList<UserAccountType>()" 
+                        class="form-control" 
+                        >
+                    <option selected>@ViewBag.AccountType</option>
```
Not quite

sakurano92 replied:
have you check comments in fix but i dont like?
are there any grammer for tag helper select to set default value with enum?
i couldnt find it on microsoft documentation, it tell me the way with no enum (use bunch of option select)
but does not tell me how to do with enum, any idea for fix?

dubeme replied:
I'll give you some hints, during next meeting.

dubeme replied:
@{
            UserAccountType activeType;

            Enum.TryParse($"{ViewData["type"]}", out activeType);

            activeType = Model?.AccountType ?? activeType;

            var accountTypes = Html.GetEnumSelectList<UserAccountType>()
                .Where(selectListItem => selectListItem.Text.Equals($"{UserAccountType.Unknown}") == false)
                .Select(selectListItem =>
                {
                    if (selectListItem.Text.Equals($"{activeType}"))
                    {
                        selectListItem.Selected = true;
                    }

                    return selectListItem;
                });
        }

        <div class="form-group col-xs-12">
            <label asp-for="AccountType"></label>
            <select asp-for="AccountType" asp-items="@accountTypes" class="form-control"></select>
        </div>

dubeme reviewed (changes requested):
Not quite what the comments meant for you to address. You removed a lot of useful code.

See comments.

sakurano92 committed:
revived unknwon in user account type

sakurano92 committed:
fix for data holds after error message

even error message is poped, now data will be kept

sakurano92 committed:
Certification Fix

Fix for certification use of prolar and singular swap name from
ViewCertification -> ViewCertifications
ViewOneCertification -> ViewCertification

sakurano92 committed:
work on to in

grammar fix as english

dubeme commented on app/wama.web/Views/AdminConsole/UserTool/ViewAccount.cshtml:
```diff
@@ -17,7 +17,7 @@
             @Html.Partial("DisplayError")
             <div style="font-size:x-large">
                 <a href="ViewAccount?memberid=@Model.MemberId"><u>Info</u></a>
-                 | <a href="ViewCertification?memberid=@Model.MemberId">  Certification
```
Remind me, I'll show you a better way to do this.

dubeme reviewed (changes requested):
Looking good, I liked what you did with the certifications. Hopefully, it makes sense to you why the name changes where necessary.

But, there's still some minor things left. See my comments.

sakurano92 committed:
Revert "Type Patron is now implementable from create user"

This reverts commit 73414ae6ca9c26ebb2b2d14f57ec45a6783cc30a.

sakurano92 committed:
reverted and added about create

sakurano92 committed:
AccountType for Create new user account page is implemented

it is implemented

sakurano92 committed:
Create new user change and view account code

dubeme reviewed (approved):
Good
